### ISSUE TITLE
contentPadding support

### ISSUE THREAD

**Issue by GuhDoy** [nanihadesuka/LazyColumnScrollbar#28]
`LazyColumn` has a param named `contentPadding` which is related to scroll bars. The current implementation doesn't take it into account, so I believe something needs to be done. I've implemented a `ViewHelper` for `RecycleView` that I think has the same logic as yours but considered padding, but I don't quite understand part of your code and don't know how to convert it. Do you have any suggestions for this?

**Comment by GuhDoy** [comment]
For your information, my implementation is here:
```
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

open class LinearLayoutViewHelper(private val list: RecyclerView) :
    BaseRecyclerViewHelper(list) {
    private val layoutManager: LinearLayoutManager = list.layoutManager as LinearLayoutManager
    private val itemCount: Int
        get() = list.adapter!!.itemCount
    private val mTempRect: Rect = Rect()

    private fun mockOffsetToReal(offset: Int): Int = mTempRect.height() * offset / mockItemSize

    private fun realOffsetToMock(offset: Int): Int = mockItemSize * offset / mTempRect.height()

    override fun getScrollRange(): Int = list.height +
            if (list.isContentScrolls()) {
                val firstIndex = layoutManager.findFirstVisibleItemPosition()
                getItemBounds(layoutManager.findViewByPosition(firstIndex)!!)
                val firstPartial = realOffsetToMock(list.paddingTop - mTempRect.top)

                val lastIndex = layoutManager.findLastVisibleItemPosition()
                getItemBounds(layoutManager.findViewByPosition(lastIndex)!!)
                val lastPartial = realOffsetToMock(mTempRect.bottom - list.height)

                realOffsetToMock(list.paddingBottom) +
                        mockItemSize * itemCount -
                        (mockItemSize * (lastIndex - firstIndex + 1) - firstPartial - lastPartial)
            } else {
                0
            }

    override fun getScrollOffset(): Int {
        val index = layoutManager.findFirstVisibleItemPosition()
        if (index == RecyclerView.NO_POSITION || index >= itemCount) {
            return 0
        }
        getItemBounds(layoutManager.findViewByPosition(index)!!)
        return index * mockItemSize - realOffsetToMock(mTempRect.top - list.paddingTop)
    }

    private fun getItemBounds(itemView: View) {
        list.getDecoratedBoundsWithMargins(itemView, mTempRect)
    }

    private var lastIndex: Int = -2
    private var lastOffset: Int = -1
    override fun scrollTo(offset: Int) {
        // Stop any scroll in progress for RecyclerView.
        list.stopScroll()
        val index = offset / mockItemSize
        val targetView = layoutManager.findViewByPosition(index)
        if (targetView != null) {
            getItemBounds(targetView)
            val realOffset = -mockOffsetToReal(offset - index * mockItemSize)
            layoutManager.scrollToPositionWithOffset(index, realOffset)
        } else {
            if (index == lastIndex - 1) {
                list.scrollBy(0, mockOffsetToReal(offset - lastOffset))
            } else {
                layoutManager.scrollToPositionWithOffset(index, 0)
            }
        }
        lastIndex = index
        lastOffset = offset
    }

    init {
        if (BuildConfig.DEBUG) {
            require(layoutManager !is GridLayoutManager || layoutManager.spanCount == 1)
        }
    }

    companion object {
        private const val mockItemSize: Int = 10000
    }
}
```

**Comment by nanihadesuka** [comment]
@GuhDoy what is the behaviour you expect for the scrollbar when one adds contentPadding to the lazy column?

Right now the code for the scrollbar does take into consideration the contentPadding but given is not an "item" of the lazy list you can't "scroll to" it with the scrollbar, nevertheless is its height is taken into account when calculating the scrollbar size.

The way is taken into account is implicit (visible lazy column height vs visible column total items height) so you won't find it mentioned directly in the code. 

Imho the lazy scrollbar doesn't need to know about the padding values of the lazy column, only the total and visible height.

**Comment by GuhDoy** [comment]
I'm using the following code (simplified) to support WindowInsets:
```
val state = rememberLazyListState()
LazyColumn(
    state = state,
    contentPadding = PaddingValues(
        bottom = with(LocalDensity.current) {
            WindowInsets.navigationBars.getBottom(this).toDp()
        }
    )
) {
    // Some items.
}
my.nanihadesuka.compose.InternalLazyColumnScrollbar(
    listState = state,
    modifier = Modifier
        .windowInsetsPadding(
            WindowInsets.systemBars
                .only(WindowInsetsSides.Horizontal + WindowInsetsSides.Bottom)
        )
)
```
Current implementation has 2 issues:

1. `LazyColumn` can still scroll backward when I drag the Scrollbar to the bottom.
![1](https://github.com/nanihadesuka/LazyColumnScrollbar/assets/46047320/af5e7adb-3105-40f2-9339-c7a000fdd594)

1. Scrollbar moves out of the screen (below) when I scroll the `LazyColumn` to the bottom.
![2](https://github.com/nanihadesuka/LazyColumnScrollbar/assets/46047320/96832618-8dbb-4a23-acd0-9c1f98e58e6d)

BTW, I've tried changing [this line of code](https://github.com/nanihadesuka/LazyColumnScrollbar/blob/98962054315d25bd4df9a877ce15ef3154eff0c2/lib/src/main/java/my/nanihadesuka/compose/LazyColumnScrollbar.kt#L160C87-L160C87) to `1f - it.visibleItemsInfo.last().fractionVisibleBottom(it.viewportEndOffset - it.afterContentPadding)` and the issues seem to be resolved, although I'm not sure if the logic is the same as the implementation I posted earlier.
